src/app/services/weather.service.ts

```diff
@@ -1,105 +1,34 @@
+   * @returns {WeatherAlertResponse} - The parsed response from the weather api
    */
-  public getLoading(): Observable<boolean> {
-    return this.loadingChanged.asObservable();
-  }
-
-  /**
-   * @description - Updates the loadingChanged BehaviorSubject. This is called within the service's call() to update the status of the API. The loading variable is updated to true when the API is called, and updated to false when the response is retrieved.
-   * @param {boolean} loading - The new value for loading
-   * @returns {void}
-   */
-  private updateLoading(loading: boolean): void {
-    this.loading = loading;
-    this.loadingChanged.next(this.loading);
-  }
-
-  /**
-   * @description - Returns the status of isSuccessfullyCompleted. This should be used in components to determine if an API should be called again.
-   * @returns {boolean} - true if the API has successfully been called; false otherwise
-   */
-  public hasSuccessfullyCompleted(): boolean {
-    return this.isSuccessfullyCompleted;
-  }
-
-  /**
-   * @description - Returns the WeatherAlert API result
-   * @returns {WeatherAlert[]} - The API results for the weather.gov api call
-   */
-  public getResults(): WeatherAlertResponse {
-    return this.apiResults;
-  }
+  protected parseApiResponse(response: any): WeatherAlertResponse {
+    return new WeatherAlertResponse(response);
+  }  
 
   /**
-   * @description - Will make a request to the weather.gov API, and retrieve weather alerts at the lat/long parameters provided. Will process the response in WeatherAlertResponse model.
-   * @param {number} lat - The latitude of the location where you'd like to request weather alerts
-   * @param {number} long - The longitude of the location where you'd like to request weather alerts
+   * @description - Takes the latitude and longitude from a job location and class the weather api with that data
+   * @param {number} lat - The latitude for the desired weather alerts
+   * @param {number} long - The longitude for teh desired weather alerts
    * @returns {void}
    */
   public call(lat: number, long: number): void {
-  // validate we're not already loading an API response and that we have the expected parameters
-    if (!this.loading && lat && long) {
-      this.updateLoading(true);
-      this.httpSubscription = this.http.get(`${this.serverURL}?point=${lat}%2C${long}&limit=500`)
-        .subscribe({
-          next: (apiResponse: any) => {
-            this.apiResults = new WeatherAlertResponse(apiResponse);
-            this.isSuccessfullyCompleted = true;
-            this.updateLoading(false);
-          },
-          error: (error: any) => {
-            this.apiResults = new WeatherAlertResponse(error.error);
-            console.error(error);
-            this.isSuccessfullyCompleted = false;
-            this.updateLoading(false);
-          }
-        });
-    }
+    this.get(`?point=${lat}%2C${long}&limit=500`);
```

Comment:
same comment here about passing in a string...lets validate the information as it used to be validated on line 87